pkg/store/bucket.go

```diff
@@ -887,6 +894,23 @@ func blockSeries(
 	return newBucketSeriesSet(res), indexr.stats.merge(chunkr.stats), nil
 }
 
+func sortLabelsForDedup(labelSet labels.Labels, replicaLabels map[string]struct{}) {
```

Comment:
Name is bit confusing to me, perhaps 
```suggestion
func moveLabelsToEnd(labelSet labels.Labels, labels map[string]struct{}) {
```